content/blog/move-resources-between-stacks/index.md

```diff
@@ -0,0 +1,187 @@
+
+The command will make no attempt to adjust the users program.  Inputs and outputs from broken dependencies will have to be adjusted in the program where the resource is being managed after being moved.  This can be done either by using [stack references](https://www.pulumi.com/docs/concepts/stack/#stackreferences), or recreating the inputs in the program.
+
+The `pulumi state move` command is available in Pulumi v3.126.0 and later.
+
+# Example
+
+To demonstrate how the command works in practice, let's go through an example.  Let's assume we have the following program:
+
+```typescript
+const randomPet = new random.RandomPet("a-random-pet");
+
+const b = new aws.s3.Bucket("b");
+
+const index = new aws.s3.BucketObject("index.html", {
+    bucket: b.bucket,
+    content: "Thanks for using Pulumi!",
+}, {
+    parent: b,
+});
+
+const randomSite = new aws.s3.BucketObject("random.html", {
+    bucket: b.bucket,
+    content: randomPet.id,
+}, {
+    parent: b,
+});
+```
+
+We create a bucket, and two bucket objects, both of them having the bucket as their parent.  One of the objects has a random pet name as its content.  At some point we decide we'd rather group all the AWS resources in a separate program, because this one grew too large.  We can do that using the `pulumi state move` command, and then adjusting the code.
+
+First we can find out which stacks already exist and we can move the resources to:
+
+```shell
+$ pulumi stack ls --all
+NAME                                            LAST UPDATE     RESOURCE COUNT  URL
+v-thomas-pulumi-corp/pulumi-demo-move-aws/dev   23 minutes ago  0               <url>
+v-thomas-pulumi-corp/pulumi-demo-move/dev       6 minutes ago   7               <url>
+```
+
+Next we can actually move the resources.  Since we're moving the resources from the currently selected stack, we can omit the `--source` argument:
+
+```
+$ pulumi state move --dest v-thomas-pulumi-corp/pulumi-demo-move-aws/dev 'urn:pulumi:dev::pulumi-demo-move::aws:s3/bucket:Bucket::b'
+Planning to move the following resources from dev to dev:
+
+  - urn:pulumi:dev::pulumi-demo-move::aws:s3/bucket:Bucket::b
+  - urn:pulumi:dev::pulumi-demo-move::aws:s3/bucket:Bucket$aws:s3/bucketObject:BucketObject::index.html
+  - urn:pulumi:dev::pulumi-demo-move::aws:s3/bucket:Bucket$aws:s3/bucketObject:BucketObject::random.html
+
+The following resources being moved to dev have dependencies on resources in dev:
+  urn:pulumi:dev::pulumi-demo-move::random:index/randomPet:RandomPet::a-random-pet has a dependency on urn:pulumi:dev::pulumi-demo-move::random:index/randomPet:RandomPet::a-random-pet
```

Comment:
I hadn't noticed before, but this doesn't quite look right.  Gonna look into fixing this.